test/e2e/clusterctl_upgrade.go

```diff
@@ -447,6 +447,9 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		})
 		Expect(workloadClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
 
+		log.Logf("Applying the cluster template yaml to the cluster in dry-run")
+		Eventually(managementClusterProxy.CreateOrUpdate, "1m", "10s").WithArguments(ctx, workloadClusterTemplate, framework.WithCreateOpts([]client.CreateOption{client.DryRunAll}...), framework.WithUpdateOpts([]client.UpdateOption{client.DryRunAll}...)).ShouldNot(HaveOccurred())
```

Comment:
Let's please pass a func into Eventually and then call CreateOrUpdate in a regular way. This seems brittle (I know once it works it works, but still...)